flocker/node/test/test_config.py

```diff
@@ -2923,6 +2976,44 @@ def test_application_with_volume_includes_max_size(self):
         }
         self.assertEqual(expected, result)
 
+    def test_application_with_volume_includes_dataset_id(self):
+        """
+        If the supplied applications has a volume with a dataset that has a
+        dataset ID, the resulting yaml will also include this dataset ID.
+        """
+        dataset_id = unicode(uuid4())
+
+        applications = [
+            Application(
+                name='mysql-hybridcluster',
+                image=DockerImage(repository='flocker/mysql', tag='v1.0.0'),
+                ports=frozenset(),
+                volume=AttachedVolume(
+                    manifestation=Manifestation(
+                        dataset=Dataset(
+                            dataset_id=dataset_id,
+                            metadata=pmap({'name': 'mysql-hybridcluster'})),
+                        primary=True),
+                    mountpoint=FilePath(b'/var/mysql/data'),
+                ),
+            )
+        ]
+        result = marshal_configuration(
+            NodeState(running=applications, not_running=[]))
+        expected = {
+            'used_ports': [],
+            'applications': {
+                'mysql-hybridcluster': {
+                    'volume': {'mountpoint': b'/var/mysql/data',
+                               'dataset_id': dataset_id},
+                    'image': u'flocker/mysql:v1.0.0',
```

Comment:
`__future__.unicode_literals` is on.  I'm growing to dislike that future feature, I guess.  It's annoying to always have to scroll up to the top of the file any time I want to know what type a string is (and seeing a '' next to a u'' always makes me want to check).